Make successful only after running twice

Good morning!
I used the following shell script to build OpenBLAS for Android.

```
NDKABI=21
NDKROOT=/home/myname/Android/Ndk/android-ndk-r15c
ROOT_DIR=`pwd`

cd "$ROOT_DIR"
cd "$ROOT_DIR/OpenBLAS-0.2.20"

ORIG_PATH="$PATH"
ORIG_CFLAGS="$CFLAGS"

export PATH="$PATH:${NDKROOT}/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin"
export CFLAGS="--sysroot=${NDKROOT}/platforms/android-${NDKABI}/arch-arm64"

make TARGET=ARMV8 BINARY=64 HOSTCC=gcc CC=aarch64-linux-android-gcc NOFORTRAN=1

export PATH="$ORIG_PATH"
export CFLAGS="$ORIG_CFLAGS"
```

When I ran the script by `sh scriptname.sh`, it fails at the first time but succeeds at the second time.

(1) `sh scriptname.sh` for the first time
```
make[1]: Leaving directory '/home/myname/Workspace/openblas-build/OpenBLAS-0.2.20/kernel'
make[1]: Entering directory '/home/myname/Workspace/openblas-build/OpenBLAS-0.2.20/exports'
perl ./gensymbol linktest  arm64 _ 0 0 1 1 0 0 "" "" 1 > linktest.c
aarch64-linux-android-gcc --sysroot=/home/myname/Android/Ndk/android-ndk-r15c/platforms/android-21/arch-arm64 -O2 -DMAX_STACK_ALLOC=2048 -Wall -DF_INTERFACE_GFORT -fPIC -DNO_LAPACK -DNO_LAPACKE -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=8 -march=armv8-a -DASMNAME= -DASMFNAME=_ -DNAME=_ -DCNAME= -DCHAR_NAME=\"_\" -DCHAR_CNAME=\"\" -DNO_AFFINITY -I. -O2 -DMAX_STACK_ALLOC=2048 -Wall -DF_INTERFACE_GFORT -fPIC -DNO_LAPACK -DNO_LAPACKE -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=8 -march=armv8-a -DASMNAME= -DASMFNAME=_ -DNAME=_ -DCNAME= -DCHAR_NAME=\"_\" -DCHAR_CNAME=\"\" -DNO_AFFINITY -I..  -shared -o ../libopenblas_armv8p-r0.2.20.so \
-Wl,--whole-archive ../libopenblas_armv8p-r0.2.20.a -Wl,--no-whole-archive \
-Wl,-soname,libopenblas.so -lm -lm
aarch64-linux-android-gcc --sysroot=/home/myname/Android/Ndk/android-ndk-r15c/platforms/android-21/arch-arm64 -O2 -DMAX_STACK_ALLOC=2048 -Wall -DF_INTERFACE_GFORT -fPIC -DNO_LAPACK -DNO_LAPACKE -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=8 -march=armv8-a -DASMNAME= -DASMFNAME=_ -DNAME=_ -DCNAME= -DCHAR_NAME=\"_\" -DCHAR_CNAME=\"\" -DNO_AFFINITY -I. -O2 -DMAX_STACK_ALLOC=2048 -Wall -DF_INTERFACE_GFORT -fPIC -DNO_LAPACK -DNO_LAPACKE -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=8 -march=armv8-a -DASMNAME= -DASMFNAME=_ -DNAME=_ -DCNAME= -DCHAR_NAME=\"_\" -DCHAR_CNAME=\"\" -DNO_AFFINITY -I..  -w -o linktest linktest.c ../libopenblas_armv8p-r0.2.20.so -L/usr/lib/gcc/x86_64-linux-gnu/5 -L/usr/lib/gcc/x86_64-linux-gnu/5/../../../x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/5/../../../../lib -L/lib/x86_64-linux-gnu -L/lib/../lib -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib -L/usr/lib/gcc/x86_64-linux-gnu/5/../../..  -lgfortran -lm -lquadmath -lm -lc   && echo OK.
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/5/libgfortran.so when searching for -lgfortran
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/5/libgfortran.a when searching for -lgfortran
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: cannot find -lgfortran
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/5/../../../x86_64-linux-gnu/libm.so when searching for -lm
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/5/../../../x86_64-linux-gnu/libm.a when searching for -lm
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/x86_64-linux-gnu/libm.so when searching for -lm
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/x86_64-linux-gnu/libm.a when searching for -lm
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/5/libquadmath.so when searching for -lquadmath
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/5/libquadmath.a when searching for -lquadmath
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: cannot find -lquadmath
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/5/../../../x86_64-linux-gnu/libm.so when searching for -lm
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/5/../../../x86_64-linux-gnu/libm.a when searching for -lm
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/x86_64-linux-gnu/libm.so when searching for -lm
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/x86_64-linux-gnu/libm.a when searching for -lm
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/5/../../../x86_64-linux-gnu/libc.so when searching for -lc
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/5/../../../x86_64-linux-gnu/libc.a when searching for -lc
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/x86_64-linux-gnu/libc.so when searching for -lc
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/x86_64-linux-gnu/libc.a when searching for -lc
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/5/libgcc.a when searching for -lgcc
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/5/../../../x86_64-linux-gnu/libc.so when searching for -lc
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/5/../../../x86_64-linux-gnu/libc.a when searching for -lc
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/x86_64-linux-gnu/libc.so when searching for -lc
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/x86_64-linux-gnu/libc.a when searching for -lc
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/5/../../../x86_64-linux-gnu/libdl.so when searching for -ldl
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/5/../../../x86_64-linux-gnu/libdl.a when searching for -ldl
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/x86_64-linux-gnu/libdl.so when searching for -ldl
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/x86_64-linux-gnu/libdl.a when searching for -ldl
/home/myname/Android/Ndk/android-ndk-r15c/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/bin/../lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/5/libgcc.a when searching for -lgcc
collect2: error: ld returned 1 exit status
Makefile:139: recipe for target '../libopenblas_armv8p-r0.2.20.so' failed
make[1]: *** [../libopenblas_armv8p-r0.2.20.so] Error 1
make[1]: Leaving directory '/home/myname/Workspace/openblas-build/OpenBLAS-0.2.20/exports'
Makefile:90: recipe for target 'shared' failed
make: *** [shared] Error 2
```

(2) `sh scriptname.sh` for the first time
```
make[1]: Leaving directory '/home/myname/Workspace/openblas-build/OpenBLAS-0.2.20/kernel'
make[1]: Entering directory '/home/myname/Workspace/openblas-build/OpenBLAS-0.2.20/exports'
make[1]: Nothing to be done for 'so'.
make[1]: Leaving directory '/home/myname/Workspace/openblas-build/OpenBLAS-0.2.20/exports'

 OpenBLAS build complete. (BLAS CBLAS)

  OS               ... Android             
  Architecture     ... arm64               
  BINARY           ... 64bit                 
  C compiler       ... GCC  (command line : aarch64-linux-android-gcc)
  Library Name     ... libopenblas_armv8p-r0.2.20.a (Multi threaded; Max num-threads is 8)

To install the library, you can run "make PREFIX=/path/to/your/installation install".
```

Would you help me? Thanks!

@martin-frbg 'cannot find -lgforttran'
I think fortran compiler is assumed from build env in final link line 
@jaewoosong
delete .so files and edit these lines to not add gfortran libraries:
```
perl ./gensymbol linktest  arm64 _ 0 0 1 1 0 0 "" "" 1 > linktest.c
aarch64-linux-android-gcc --sysroot=/home/myname/Android/Ndk/android-ndk-r15c/platforms/android-21/arch-arm64 -O2 -DMAX_STACK_ALLOC=2048 -Wall -DF_INTERFACE_GFORT -fPIC -DNO_LAPACK -DNO_LAPACKE -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=8 -march=armv8-a -DASMNAME= -DASMFNAME=_ -DNAME=_ -DCNAME= -DCHAR_NAME=\"_\" -DCHAR_CNAME=\"\" -DNO_AFFINITY -I. -O2 -DMAX_STACK_ALLOC=2048 -Wall -DF_INTERFACE_GFORT -fPIC -DNO_LAPACK -DNO_LAPACKE -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=8 -march=armv8-a -DASMNAME= -DASMFNAME=_ -DNAME=_ -DCNAME= -DCHAR_NAME=\"_\" -DCHAR_CNAME=\"\" -DNO_AFFINITY -I..  -shared -o ../libopenblas_armv8p-r0.2.20.so \
-Wl,--whole-archive ../libopenblas_armv8p-r0.2.20.a -Wl,--no-whole-archive \
-Wl,-soname,libopenblas.so -lm -lm
aarch64-linux-android-gcc --sysroot=/home/myname/Android/Ndk/android-ndk-r15c/platforms/android-21/arch-arm64 -O2 -DMAX_STACK_ALLOC=2048 -Wall -DF_INTERFACE_GFORT -fPIC -DNO_LAPACK -DNO_LAPACKE -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=8 -march=armv8-a -DASMNAME= -DASMFNAME=_ -DNAME=_ -DCNAME= -DCHAR_NAME=\"_\" -DCHAR_CNAME=\"\" -DNO_AFFINITY -I. -O2 -DMAX_STACK_ALLOC=2048 -Wall -DF_INTERFACE_GFORT -fPIC -DNO_LAPACK -DNO_LAPACKE -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=8 -march=armv8-a -DASMNAME= -DASMFNAME=_ -DNAME=_ -DCNAME= -DCHAR_NAME=\"_\" -DCHAR_CNAME=\"\" -DNO_AFFINITY -I..  -w -o linktest linktest.c ../libopenblas_armv8p-r0.2.20.so -L/usr/lib/gcc/x86_64-linux-gnu/5 -L/usr/lib/gcc/x86_64-linux-gnu/5/../../../x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/5/../../../../lib -L/lib/x86_64-linux-gnu -L/lib/../lib -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib -L/usr/lib/gcc/x86_64-linux-gnu/5/../../..  -lgfortran -lm -lquadmath -lm -lc   && echo OK.
```

--- a/exports/Makefile
+++ b/exports/Makefile
@@ -51,6 +51,10 @@ endif
 endif
 endif
 
+ifeq ($(NOFORTRAN), $(filter $(NOFORTRAN),1 2))
+FEXTRALIB =
+endif
+
 all::
 
 libs::

